lib/dock.rb

```diff
@@ -0,0 +1,35 @@
+class Dock
+  attr_reader :name, :max_rental_time, :current_rentals, :rental_history
+
+  def initialize(name, max_rental_time)
+    @name = name
+    @max_rental_time = max_rental_time
+    @current_rentals = []
```

Comment:
If you were storing rentals in a hash, rather than a nested array, it might make your return method a little cleaner.  I'm imagining something like `@current_rentals = { boat => renter }` where your keys are specific instances of the boat class.